packages/uhk-web/src/app/components/macro/action-editor/tab/key/macro-key.component.ts

```diff
@@ -71,4 +73,9 @@ export class MacroKeyTabComponent implements OnInit {
         return keyMacroAction;
     }
 
+    isMacroValid = () => {
+        const keyMacroAction = this.getKeyMacroAction();
+        return keyMacroAction.scancode !== undefined || keyMacroAction.modifierMask !== undefined;
```

Comment:
pls use the following condition `!!keyMacroAction.scancode || !!keyMacroAction.modifierMask;`

@mondalaci If I read well the specification
>Key macro action: Scancode None and no modifiers selected

then the condition is && instead of ||

Reply:
@ert78gb It must be disabled if scancode is none and no modifiers is selected. Thus, it's be valid (not disabled) if scancode is not none or any modifiers are selected. The current implementation is correct.

Reply:
thank you guys.